File: VB/DXDocumentViewerExternalRibbon/Controllers/HomeController.vb
Imports System.Web.Mvc
Imports DevExpress.Web.Mvc

' ...
Namespace DXDocumentViewerExternalRibbon.Controllers

    Public Class HomeController
        Inherits Controller

        Private ReadOnly report As Reports.XtraReport1 = New Reports.XtraReport1()

        Public Function Index() As ActionResult
            Return View()
        End Function

        Public Function ReportPartial() As ActionResult
            Return PartialView("_ReportPartial", report)
        End Function

        Public Function ReportPartialExport() As ActionResult
            Return DocumentViewerExtension.ExportTo(report, Request)
        End Function
    End Class
End Namespace

File: CS/DXDocumentViewerExternalRibbon/Controllers/HomeController.cs
using System.Web.Mvc;
using DevExpress.Web.Mvc;
// ...

namespace DXDocumentViewerExternalRibbon.Controllers {
    public class HomeController : Controller {
        readonly DXDocumentViewerExternalRibbon.Reports.XtraReport1 report = 
            new DXDocumentViewerExternalRibbon.Reports.XtraReport1();

        public ActionResult Index() {
            return View();
        }

        public ActionResult ReportPartial() {
            return PartialView("_ReportPartial", report);
        }

        public ActionResult ReportPartialExport() {
            return DocumentViewerExtension.ExportTo(report, Request);
        }
    }
}
